client/app/components/services/service-details/service-details.component.js

```diff
@@ -284,11 +284,15 @@ function ComponentController($state, $window, CollectionsApi, EventNotifications
   }
 
   function openConsole(item) {
-    Consoles.open(item.id);
+    if (item['supports_console?'] && item.power_state === 'on') {
+      Consoles.open(item.id);
+    }
   }
 
   function openCockpit(item) {
-    $window.open('http://' + item.ipaddresses[0] + ':9090');
+    if (item['supports_cockpit?'] && item.power_state === 'on') {
+      $window.open('http://' + item.ipaddresses[0] + ':9090');
```

Comment:
I wouldn't know where to make this optional.  Our existing cockpit URL is like this so I am not sure I would be inclined to just change it.